src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics.X86;
+
+namespace System
+{
+    internal static class BitOps
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int TrailingZeroCount(int matches)
+        {
+            if (Bmi1.IsSupported)
+            {
+                return (int)Bmi1.TrailingZeroCount((uint)matches);
+            }
+            else
+            {
+                if (BitConverter.IsLittleEndian)
+                {
+                    return TrailingZeroCountFallback(matches);
+                }
+                else
+                {
+                    // Not sure the above failback will work on BigEndian
+                    Debug.Fail("TrailingZeroCount not implemented for BigEndian");
+                    return 0;
+                }
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int TrailingZeroCountFallback(int matches)
```

Comment:
I think this is small enough we could probably just have it as a local function and the JIT should reliably inline it (especially with @AndyAyersMS change here: https://github.com/dotnet/coreclr/pull/21893).

Reply:
or, given that it is a single line, we probably don't even need the local function.